librz/core/casm.c

```diff
@@ -219,6 +219,36 @@ RZ_API RzCmdStatus rz_core_asm_plugins_print(RZ_NONNULL RZ_BORROW RzCore *core,
 	return RZ_CMD_STATUS_OK;
 }
 
+RZ_API RzCmdStatus rz_core_cpu_descs_print(RZ_NONNULL RzCore *core, RZ_NONNULL const char *plugin) {
+	rz_return_val_if_fail(core && plugin, RZ_CMD_STATUS_ERROR);
+	RzAsm *a = core->rasm;
+	RzIterator *iter = ht_sp_as_iter(a->plugins);
+	RzList *plugin_list = rz_list_new_from_iterator(iter);
+	if (!plugin_list) {
+		rz_iterator_free(iter);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_list_sort(plugin_list, (RzListComparator)rz_asm_plugin_cmp, NULL);
+	RzListIter *it;
+	RzAsmPlugin *ap;
+	rz_list_foreach (plugin_list, it, ap) {
+		if (ap->cpus && !strcmp(plugin, ap->name)) {
+			char **desc = ap->get_cpu_desc();
+			if (desc) {
+				for (size_t i = 0; desc[i] != NULL; i += 2) {
+					rz_cons_printf("%-15s %s", desc[i], desc[i + 1]);
+					rz_cons_newline();
+				}
+				break;
+			} else {
+				return RZ_CMD_STATUS_ERROR;
+			}
```

Comment:
```suggestion
			if (!desc) {
				rz_iterator_free(iter);
				return RZ_CMD_STATUS_ERROR;
			}
			for (size_t i = 0; desc[i] != NULL; i += 2) {
				rz_cons_printf("%-15s %s", desc[i], desc[i + 1]);
				rz_cons_newline();
			}
			break;
```

Also free the `plugin_list` below and above in case of `!desc`.

Also, the inverted condition is nice. Because we have less indentations and it is easier to read. A nice trick in general.